<launch>
  <arg name="ip"/>
  <arg name="port" default="8899"/>
  <arg name="debug" default="false"/>

  <env if="$(arg debug)" name="ROSCONSOLE_CONFIG_FILE" value="$(find aubo_hardware_interface)/rosconsole.conf"/>

  <rosparam command="load" file="$(find aubo_hardware_interface)/config/hardware.yaml" />
  <rosparam command="load" file="$(find aubo_hardware_interface)/config/controllers.yaml" />

  <!-- -->
  <node name="aubo_i7" pkg="aubo_hardware_interface" type="aubo_hw" output="screen">
    <param name="tcp/hostname" value="$(arg ip)"/>
    <param name="tcp/port" value="$(arg port)"/>
    <rosparam ns="aubo" command="load" file="$(find aubo_hardware_interface)/config/aubo.yaml"/>
  </node>

  <!-- -->
  <node ns="aubo_i7" name="controller_spawner" pkg="controller_manager" type="spawner" output="screen"
    args="/aubo_i7/controller/state
          /aubo_i7/controller/arm_controller"/>

</launch>
